Source/DafnyCore/Compilers/Compiler-python.cs

```diff
@@ -271,7 +272,9 @@ public class PythonCompiler : SinglePassCompiler {
       var wDefault = btw.NewBlockPy($"def default(cls, {UsedTypeParameters(dt).Comma(FormatDefaultTypeParameterValue)}):");
       var groundingCtor = dt.GetGroundingCtor();
       if (groundingCtor.IsGhost) {
-        wDefault.WriteLine($"return {ForcePlaceboValue(UserDefinedType.FromTopLevelDecl(dt.tok, dt), wDefault, dt.tok)}");
+        wDefault.WriteLine($"return lambda: {ForcePlaceboValue(simplifiedType, wDefault, dt.tok)}");
```

Comment:
We only use `simplifiedType` here, but we don't even need it, because `ForcePlaceboValue` does the same step for us.
```suggestion
        wDefault.WriteLine($"return lambda: {ForcePlaceboValue(UserDefinedType.FromTopLevelDecl(dt.tok, dt), wDefault, dt.tok)}");
```